test/ServerTests/Diagnostics/DiagnosticsStatusCodesMvcTests.cs

```diff
@@ -0,0 +1,138 @@
+        }
+
+        private async Task NonExistingPage(ServerType server, RuntimeFlavor runtimeFlavor, RuntimeArchitecture architecture, string applicationBaseUrl)
+        {
+            await ServerHelper.ServerResponseMessageTest(
+                SiteName,
+                server,
+                runtimeFlavor,
+                architecture,
+                applicationBaseUrl,
+                async (httpClient, logger, token) =>
+                {
+                    var response = await RetryHelper.RetryRequest(async () =>
+                    {
+                        return await httpClient.GetAsync("/bob");
+                    }, logger, token);
+
+                    var responseText = await response.Content.ReadAsStringAsync();
+                    Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
+                });
+        }
+
+        [ConditionalTheory]
+        [OSSkipCondition(OperatingSystems.Linux)]
+        [OSSkipCondition(OperatingSystems.MacOSX)]
+        [InlineData(RuntimeFlavor.Clr, RuntimeArchitecture.x86, "http://localhost:6112")]
+        [InlineData(RuntimeFlavor.Clr, RuntimeArchitecture.x64, "http://localhost:6113")]
+        [InlineData(RuntimeFlavor.CoreClr, RuntimeArchitecture.x86, "http://localhost:6114")]
+        [InlineData(RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, "http://localhost:6115")]
+        public async Task CustomErrorCode_WebListener(RuntimeFlavor runtimeFlavor, RuntimeArchitecture architecture, string applicationBaseUrl)
+        {
+            await CustomErrorCode(ServerType.WebListener, runtimeFlavor, architecture, applicationBaseUrl);
+        }
+
+        [ConditionalTheory]
+        [InlineData(RuntimeFlavor.Clr, RuntimeArchitecture.x64, "http://localhost:6116")]
+        [InlineData(RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, "http://localhost:6117")]
+        public async Task CustomErrorCode_Kestrel(RuntimeFlavor runtimeFlavor, RuntimeArchitecture architecture, string applicationBaseUrl)
+        {
+            await CustomErrorCode(ServerType.Kestrel, runtimeFlavor, architecture, applicationBaseUrl);
+        }
+
+        public async Task CustomErrorCode(ServerType server, RuntimeFlavor runtimeFlavor, RuntimeArchitecture architecture, string applicationBaseUrl)
+        {
+            await ServerHelper.ServerResponseMessageTest(
+                SiteName,
+                server,
+                runtimeFlavor,
+                architecture,
+                applicationBaseUrl,
+                async (httpClient, logger, token) =>
+                {
+                    var response = await RetryHelper.RetryRequest(async () =>
+                    {
+                        return await httpClient.GetAsync("/errors/" + (int)HttpStatusCode.PaymentRequired);
+                    }, logger, token);
+
+                    var responseText = await response.Content.ReadAsStringAsync();
```

Comment:
Try/Catch&Log the response text if the Assert fails. Otherwise it's hard to debug tests on the CI. See https://github.com/aspnet/IISIntegration/blob/dev/test/Microsoft.AspNet.IISPlatformHandler.FunctionalTests/NtlmAuthentationTest.cs#L116-L121